content/5_Plat/HLD.mdx

```diff
@@ -15,6 +15,218 @@ frequency: 1
 
 ## Tutorial
 
+In some advanced problems on trees, the problem may ask you to support the following operations:
+
+- Update all nodes along the path from node $x$ to node $y$. (Note that this path is unique due to properties of trees)
+
+- Find the sum, maximum, minimum (or any other operation that satisfies the associative property) along the path from node $x$ to node $y$.
+
+Other simpler algorithms such as difference on trees or LCAs could be used to support any one of the operations. However, Heavy Light Decomposition (or HLD) is needed to support both. 
+
+**Definitions:**
+
+To understand HLD, I will first clarify the definition of these important terms: 
+
+- A  **heavy child** of a node is the child with the largest subtree size rooted at the child
+- A  **light child** of a node is any child that is not a **heavy child**
+- A  **heavy edge** connects the current node to its **heavy child**
+- A  **light edge** connects the current node to any of its **light children**
+- A  **heavy path** is the path formed by a collection **heavy edges**
+- A  **light path** is the path formed by a collection **light edges**
+
+**Properties:**
+
+Given the definitions stated above, an important property can be derived (I will prove this a bit later).
+
+- Any path from node $x$ to node $y$ on the tree can pass through at most $O(logN)$ light edges.
+
+Lets suppose the above property is true for a moment. Since a heavy path can only be broken by a light edge (or else the edge will be a part of the heavy path), we can know that there are at most $O(logN)$ heavy paths on any path from an arbitrary node $x$ to an arbitrary node $y$.
```

Comment:
There are a lot of formatting issues. Time complexity should be formatted as given here ($\mathcal O$, $\log$):

https://usaco.guide/general/adding-solution?lang=cpp#steps

not all variables are surrounded by $$, some sentences don't end with periods. Also a typo I noticed (though I didn't look very carefully)

> propert

someone else can review the content